infrastructure/containers/postgres/automatic-backup/backup_or_restore.py

```diff
@@ -129,14 +292,41 @@ def restore_database(backup_fpath):
         user=os.environ["POSTGRES_USER"],
         db_name=os.environ["POSTGRES_DB"],
     )
-    run_shell_command(create_db_cmd, env_vars=pg_env_vars)
+    run_shell_command(command=create_db_cmd, env_vars=pg_env_vars)
+
+    # find the most recent backup or verify the specify backup exists
+    print("Getting backup file from S3")
+    session = create_s3_session()
+
+    if not backup_object_name_to_restore_from__override:
```

Comment:
Sorry, you must be getting sick of me picking on this same spot over and over. So, there are two principles of program design that this whole function is currently violating:

1. It's really long. Functions should be short and specialized.
2. Not every bit of code in this function is on the same level of abstraction. Your function `download_backup_object()` is perfect. The fact that all we do here is call that is amazing. This function shouldn't need to worry about the exact details of how that is happening. I would say that this big if/else/elif statement qualifies as more detailed than this function needs to know about. And when I look at it after not having seen it in awhile, I'm afraid to go deeper.

I recommend extracting this if/else statement into a separate function. We could call it `get_s3_key_of_backup_to_restore_from() -> str` or `get_s3_object_name_of_backup_to_restore_from`() -> str. You could have that function take whatever inputs it needs and even raise an error.

Is this pedantic? Yes :( But I promise the concept has value and leads to less buggy code over time.